torch_audiomentations/augmentations/pitch_shift.py

```diff
@@ -0,0 +1,73 @@
+from random import choices
+import torch
+import torch.nn.functional as F
+
+from ..core.transforms_interface import BaseWaveformTransform
+from torch_pitch_shift import *
+
+
+class PitchShift(BaseWaveformTransform):
+    """
+    Pitch-shift the sound up or down without changing the tempo
+    """
+
+    supports_multichannel = True
+    requires_sample_rate = True
+
+    def __init__(
+        self,
+        sample_rate: int,
+        min_transpose_ratio=0.5,
+        max_transpose_ratio=2,
+        mode: str = "per_example",
+        p: float = 0.5,
+        p_mode: str = None,
+    ):
+        """
+        :param min_transpose_ratio: Minimum pitch shift transposition ratio (default 0.5 --> -1 octaves)
+        :param max_transpose_ratio: Maximum pitch shift transposition ratio (default 2 --> +1 octaves)
+        :param mode:
+        :param p:
+        :param p_mode:
+        :param sample_rate:
+        """
+        super().__init__(mode, p, p_mode, sample_rate)
+
+        self._min_transpose_ratio = min_transpose_ratio
+        self._max_transpose_ratio = max_transpose_ratio
+        if self._min_transpose_ratio > self._max_transpose_ratio:
+            raise ValueError("max_transpose_ratio must be > min_transpose_ratio")
+        self._pitch_shift = PitchShifter()
+        self._sample_rate = sample_rate
```

Comment:
At this point we need to assert that a valid (not None) sample rate is specified